src/lib/useAddressHistory.ts

```diff
@@ -17,7 +17,11 @@ export const useAddressHistoryStore = create<AddressHistoryState>()(
     (set, get) => ({
       addresses: [],
       addAddress: (address: string) => {
-        if (get().addresses.includes(address)) return;
+        if (get().addresses.includes(address)) {
+          return set({
+            addresses: [address, ...get().addresses.filter((a) => a !== address)].slice(0, 20),
+          });
+        }
         return set({ addresses: [address, ...get().addresses].slice(0, 20) });
```

Comment:
```suggestion
        return set({
            addresses: [address, ...get().addresses.filter((a) => a !== address)].slice(0, 20),
          });
```

Won't this be more elegant?